lib/account/account-manager.js

```diff
@@ -0,0 +1,101 @@
+"use strict";
+/**
+ * Manages the business logic of account controls.
+ */
+
+/**
+ * The object sent in which has information needed to validate the user
+ * has properly set up their two factor authentication and has generated
+ * a password for use to save.
+ *
+ * @typedef {Account~accountinfo}
+ * @param {string} accountId this is the unhashed account id sent to client
+ * @param {string} currentMfa
+ * @param {string} previousMfa
+ * @param {string} password
+ */
+class AccountManager {
+    /**
+     * @param {Object} accountService
+     * @param {Object} config
+     * @param {Object} hotp
+     * @param {Object} otDate
+     * @param {Object} password
+     * @param {Object} promise
+     * @param {Object} random
+     */
+    constructor(accountService, config, hotp, otDate, random, promise) {
+        this.accountService = accountService;
+        this.config = config;
+        this.hotp = hotp;
+        this.otDate = otDate;
+        this.random = random;
+        this.promise = promise;
+    }
+
+
+    /**
+     * Validates the previous and current MFA values are valid and sends
+     * data off to Account service class for updating/putting data into
+     * the Account record.
+     *
+     * Password generation should follow this format:
+     * hash(hash(password) + salt)
+     *
+     * @param {AccountManager~accountInfo} accountInfo
+     * @return {Promise}
+     */
+    complete(accountInfo) {
+        return this.accountService.get(accountInfo.accountId).then((accountFile) => {
```

Comment:
I would not want this method to pass the accountFile back into complete.  Can the accountService cache the accountFile results instead and use those for future requests?  Caching for only the lifetime of the request, not longer.